From 1037099b002f8621b4bc7de014b84957eb624b46 Mon Sep 17 00:00:00 2001
From: Lukas Bezdicka <lbezdick@redhat.com>
Date: Wed, 17 Oct 2018 00:10:26 +0200
Subject: [PATCH] Standby the pacemaker on node before taking it down

During testing I found APIs become unavailable because of pcs
cluster stop command. This can be avoided by gracefull standby.
Other change here is adding --wait=300 to yum_update pcs commands.
Turns out under heavy load this becomes racy and cluster is still
running while we yum update.

Change-Id: I939464b93df2c33aa662a1598c8b4f478b798ec2
---
 extraconfig/tasks/yum_update.sh | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/extraconfig/tasks/yum_update.sh b/extraconfig/tasks/yum_update.sh
index 66d0242f7..338f136ff 100755
--- a/extraconfig/tasks/yum_update.sh
+++ b/extraconfig/tasks/yum_update.sh
@@ -105,7 +105,8 @@ if [[ "$pacemaker_status" == "active" ]] ; then
         echo "Active node count is 1, stopping node with --force"
         pcs cluster stop --force
     else
-        pcs cluster stop
+        pcs node standby --wait=300
+        pcs cluster stop --wait=300
     fi
     update_network
 else
@@ -131,7 +132,8 @@ echo "yum return code: $return_code"
 
 if [[ "$pacemaker_status" == "active" ]] ; then
     echo "Starting cluster node"
-    pcs cluster start
+    pcs cluster start --wait=300
+    pcs node unstandby --wait=300
 
     hostname=$(hostname -s)
     tstart=$(date +%s)
-- 
2.17.2

